christfellowshipchurch/web-app
# [LiveStream] LiveStream event join/close interactions

# Related Issue
- Closes #215

# Changes or Updates
- Dispatches `LIVESTREAM_JOINED` interaction when an authenticated user loads a live streaming event
- Dispatches `LIVESTREAM_CLOSED` interaction when an authenticated user navigates away from or closes a live streaming event page
- Fixes a transparent/missing background behind Event Chat panel when loading and/or error

# Test Instructions
- Open the [interactions page on Rock](https://rock.christfellowship.church/page/664?ComponentId=181648) for live events (interactions `ComponentId=181648`)
- Be logged in as a user with access to the test live stream events/feature flags
- Load a local live event like: http://local.christfellowship.church:3000/events/test-event-24-hour-live-stream
- [x] Verify you see a `LIVESTREAM_JOINED` event for your user
- Navigate away from the page
- [x] Verify you see a `LIVESTREAM_CLOSED` event for your user
- Go back to the event, re-verify the new `JOINED` event, then close the entire page/window/tab
- [x] Verify you still see a `LIVESTREAM_CLOSED` event for your user

# Screenshots
|Caption|Screenshot|
|---|---|
|Going back and forth to event page via navigation and closing the tab|<img src="https://user-images.githubusercontent.com/1812955/101385209-76382c80-3889-11eb-9a0b-11600625033b.jpg" width="640" />|
|Joined interaction detail|<img src="https://user-images.githubusercontent.com/1812955/101385220-7a644a00-3889-11eb-9410-d69aec513f9d.jpg" width="520" />|
|Closed interaction detail|<img src="https://user-images.githubusercontent.com/1812955/101385216-79cbb380-3889-11eb-9eda-1e9e48c115c2.jpg" width="520" />|
|Fix for chat panel background (before, was transparent)|<img src="https://user-images.githubusercontent.com/1812955/101385539-ef378400-3889-11eb-833b-89b704439bab.jpg" width="520" />|



# Tests
